### PR TITLE
docs: addition of Documentation.rst

### PR DESCRIPTION
**Pull request by dinosk**
Signed-off-by: Dinos Kousidis <konstantinos.kousidis@cern.ch>

### CODE REVIEW

**Comment by lnielsen** [comment]
I think you need to split this document up into some pieces.....some shoudl go in README.rst, others should perhaps go in an "Overview" section or similar (see e.g. https://github.com/inveniosoftware/invenio-access/blob/master/docs/overview.rst and invenio-access.readthedocs.io). Perhaps let's discuss IRL what is the best structure so we have common practices over all modules.

**Comment by dinosk** [comment]
Yes sure. I thought of writing this as a guide on how to add new statistics which explains the architecture of the module (entrypoints, events+aggs+queries) in the process.

File: README.rst [hunk 1]
```diff
@@ -40,10 +40,19 @@
...
 Invenio module for collecting statistics.
 
-*This is an experimental developer preview release.*
+This module provides the components for **statistical data processing and
+querying**.
+
+The most common statistics measure the occurence of events in an invenio
+application, e.g. file downloads, record views and others. Invenio-stats
+provides the tools to transform, register, compress and query those events.
+However statistics can be fully customized and directly query the database.
```
Comment (dinosk) [inline README.rst]: However,

File: docs/configuration.rst [hunk 2]
```diff
@@ -25,5 +25,46 @@
 Configuration
 =============
 
-.. automodule:: invenio_stats.config
-   :members:
+
+Events creation
+---------------
+The creation of events can be done via signals. Invenio-stats provide an
```
Comment (dinosk) [inline docs/configuration.rst]: provides (in a previous sentence invenio-stats is considered singural)

File: docs/configuration.rst [hunk 3]
```diff
@@ -25,5 +25,46 @@
...
+.. code-block:: python
+    from datetime import timedelta
+    CELERY_BEAT_SCHEDULE = {
+        'indexer': {
+            'task': 'invenio_stats.tasks.process_events',
+            'schedule': timedelta(hours=3),
+        },
+    }
+
+Invenio stats provide aggregations
```
Comment (dinosk) [inline docs/configuration.rst]: I don't understand this line

File: docs/overview.rst [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+    MA 02111-1307, USA.
+
+    In applying this license, CERN does not
+    waive the privileges and immunities granted to it by virtue of its status
+    as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+
+Overview
+--------
+Invenio-Stats provides enables an Invenio instance to generate statistics and
```
Comment (dinosk) [inline docs/overview.rst]: no need for provides

File: docs/overview.rst [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+access them via a REST API.
+
+
+1. Event processing
+~~~~~~~~~~~~~~~~~~~
+Statistics can measure the occurence of event within the application (
```
Comment (dinosk) [inline docs/overview.rst]: of (an) event or of events

File: docs/overview.rst [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+the entrypoint `invenio_stats.events`.
+
+.. code-block:: python
+ 
+ 'invenio_stats.events': [
+     'invenio_stats = '
+     'invenio_stats.contrib.registrations:register_events'
+ ]
+
+The function enables to dynamically add new event using for example
```
Comment (dinosk) [inline docs/overview.rst]: events or a new event

File: docs/overview.rst [hunk 7]
```diff
@@ -0,0 +1,361 @@
...
+                double_click_window=30,
+                # Create one index per month which will store file download events
+                suffix='%Y-%m'
+            ))
+    }
+
+1.2. Emitting Signals (optional)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The above diagram shows a **signal** being monitored. This signal occurences
```
Comment (dinosk) [inline docs/overview.rst]: These signal occurences or The signal occurences

File: docs/overview.rst [hunk 8]
```diff
@@ -0,0 +1,361 @@
...
+are registered as **Events**. Invenio modules already send many different
+signals. If no signal exist it is possible to emit your own signals.
```
Comment (dinosk) [inline docs/overview.rst]: exists

File: docs/overview.rst [hunk 9]
```diff
@@ -0,0 +1,361 @@
...
+
+    current_stats.publish('my-event-name', [event])
+
+
+1.4. Event Processing
+^^^^^^^^^^^^^^^^^^^
+Now that an event is recorded, the next step is adding it to our Elasticsearch
+storage so that the new statistics can be calculated. A default event indexer
+for this task is provided in processors.py. An indexer is assigned to each
+event as seen in section 1, and also a list preprocessors is given. These are
```
Comment (dinosk) [inline docs/overview.rst]: list of preprocessors

File: docs/overview.rst [hunk 10]
```diff
@@ -0,0 +1,361 @@
...
+                ),
+                required_filters=dict(
+                    bucket_id='bucket_id',
+                    file_key='file_key',
+                )
+            )
+        )
+   ...
+
+The logic is identical, we specify the query class, each for a given statistic. The document type specifies the Elasticsearch doc_type from where we will get the results, the copy fields function in the same as in queries, and the required filters are the parameters which must be given in order to identify the resource.
```
Comment (dinosk) [inline docs/overview.rst]: the copy fields function is the same as in queries

File: invenio_stats/__init__.py [hunk 11]
```diff
@@ -22,7 +22,19 @@
...
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-"""Invenio module for collecting statistics."""
+"""Invenio module for collecting statistics.
+
+Those events are
+often sent via signals which are then recorded by invenio-stats listeners.
+Those events can also be transformed, .
+
+Invenio-stats provides the tools :
```
Comment (dinosk) [inline invenio_stats/__init__.py]: tools to

**Pull request merged**